File: 240622/Run Length 인코딩/run-length-encoding.py
input_str = input()

def run_length_encoding(input_str):
    ret_str = ""
    
    indices = [-1]
    for i in range(len(input_str) - 1):
        if input_str[i] != input_str[i + 1]:
            indices.append(i)
    indices.append(len(input_str) - 1)
    
    cnts_lst = []
    for i in range(len(indices) - 1):
        diff = indices[i + 1] - indices[i]
        cnts_lst.append(diff)

    for chr_idx, cnts in zip(indices[1:], cnts_lst):
        ret_str += (input_str[chr_idx] + str(cnts))

    return ret_str

answer = run_length_encoding(input_str)
print(len(answer))
print(answer)

File: 251202/Run Length 인코딩/run-length-encoding.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

string run_length_encoding(const string& input_str) 
{
    string ret_str;

    int n = input_str.length();
    if (n == 0) return ret_str; 

    vector<int> indices;
    indices.push_back(-1);

    for (int i = 0; i < n - 1; i++) 
    {
        if (input_str[i] != input_str[i + 1]) 
        {
            indices.push_back(i);
        }
    }
    indices.push_back(n - 1);

    vector<int> cnts_lst;
    for (int i = 0; i < (int)indices.size() - 1; i++) 
    {
        int diff = indices[i + 1] - indices[i];
        cnts_lst.push_back(diff);

    for (int i = 0; i < (int)cnts_lst.size(); i++) 
    {
        int chr_idx = indices[i + 1];
        int cnts = cnts_lst[i];
        ret_str += input_str[chr_idx];
        ret_str += to_string(cnts);
    }

    return ret_str;
}

int main() {
    string input_str;
    getline(cin, input_str);  // 공백 포함 문자열 입력

    string answer = run_length_encoding(input_str);
    cout << answer.length() << "\n";
    cout << answer << "\n";

    return 0;
}
